model/things/src/main/java/org/eclipse/ditto/model/things/ThingPolicyIdInvalidException.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2017-2018 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.model.things;
+
+import java.net.URI;
+import java.text.MessageFormat;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.model.base.common.HttpStatusCode;
+import org.eclipse.ditto.model.base.exceptions.DittoRuntimeException;
+import org.eclipse.ditto.model.base.exceptions.DittoRuntimeExceptionBuilder;
+import org.eclipse.ditto.model.base.headers.DittoHeaders;
+import org.eclipse.ditto.model.base.json.JsonSchemaVersion;
+
+/**
+ * Thrown if the ID set as policyId of the {@link Thing} is not valid
+ * (for example if it does not comply to the Entity ID REGEX).
+ */
+@Immutable
+public final class ThingPolicyIdInvalidException extends DittoRuntimeException implements ThingException {
+
+    /**
+     * Error code of this exception.
+     */
+    public static final String ERROR_CODE = ERROR_CODE_PREFIX + "policyId.invalid";
```

Comment:
ah, I just saw that the error code is camel cased here - AFAIR we never camel cased error codes - I would rather use `"policyid.invalid"` (similar to the `FeatureIdInvalidException`)

Reply:
Oh good catch, sorry. Fixed it